baseplate/server/__init__.py

```diff
@@ -433,22 +444,43 @@ def load_and_run_shell() -> None:
 
         ipython_config = Config()
         ipython_config.TerminalInteractiveShell.banner2 = banner
-        start_ipython(argv=[], user_ns=env, config=ipython_config)
+        start_ipython(argv=[f"--logfile=/{console_logpath}"], user_ns=env, config=ipython_config)
         raise SystemExit
     except ImportError:
         import code
 
         newbanner = f"Baseplate Interactive Shell\nPython {sys.version}\n\n"
         banner = newbanner + banner
 
-        # import this just for its side-effects (of enabling nice keyboard
-        # movement while editing text)
         try:
-            import readline
+            _InteractiveConsole_setup(env, console_logpath)
 
-            del readline
         except ImportError:
             pass
 
         shell = code.InteractiveConsole(locals=env)
         shell.interact(banner)
+
+
+def _shell_commands_log_path() -> str:
+    # Define path for console log output
+    pid_1_path = "/proc/1/fd/1"
+    # check if running in a containerized environment
+    if os.getenv("KUBERNETES_SERVICE_HOST") and os.access(pid_1_path, os.W_OK):
+        return os.path.abspath(pid_1_path)
+    else:
+        return os.path.abspath("/var/log/.shell_history")
+
+
+def _InteractiveConsole_setup(env: Dict, console_logpath: str) -> None:
+    # Setup some quality of life console interactions
+    import readline
+
+    readline.set_completer(Completer(env).complete)
+    readline.parse_and_bind("tab: complete")
+
+    # Define audit logging with readline history
+    def save_console_history(history_path=console_logpath) -> None:
+        readline.write_history_file(history_path)
```

Comment:
Do you want to `set_history_length` so it doesn't grow unruly?

Reply:
:no_bell: It seems like if you don't `read_history_file` then `write_history_file` will overwrite the file. Is that ok for auditing purposes?

Reply:
I'd like to see an unlimited history size, but it's probably more prudent to set a high upper bounds. Unlikely that a human would have _that_ many commands in a history. 

As for `read_history_file`, I opted to not include that because this output should be streamed out and not rely on local disk storage. Unlikely that a concurrency issue would arise if running not in a containerized environment.